src/main/java/org/bihealth/mi/easybus/implementations/email/BusEmail.java

```diff
@@ -107,7 +106,33 @@ public boolean isAlive() {
 
     @Override
     public void send(Message message, Scope scope, Participant participant) throws BusException {
-        this.connection.send(message, scope, participant);
+        // Init
+        boolean sent = false;
+        int tryCounter = 1;
+        
+        // Try to send e-mail
+        while (!sent) {
+            try {
+                this.connection.send(message, scope, participant);
+                sent = true;
+            } catch (BusException e) {
+
+                // Throw exception after threshold
+                if (tryCounter == Resources.MAX_TRY_SEND_MAIL) {
+                    throw new BusException("Unable to send e-mail", e);
+                }
+
+                // Add counter
+                tryCounter++;
+
+                // Sleep
+                try {
+                    Thread.sleep(threadSleepTime);
+                } catch (InterruptedException e1) {
+                    throw new BusException("Unable to send e-mail due to interrupted thread", e1);
```

Comment:
This might have unintended side effects when a Bus is stopped while a send operation is being tried. Wouldn't it be better to throw the InterruptedException? Probably, yes. Please analyze and re-consider.

Reply:
I dont understand since an exception is thrown: You mean throw the actual InterruptedException?